lib/src/main/java/com/cloudbees/groovy/cps/impl/CpsCallableInvocation.java

```diff
@@ -64,6 +71,23 @@ void checkMismatch(Object expectedReceiver, List<String> expectedMethodNames) {
             return; 
         }
 
+        if (expectedReceiver instanceof GroovyObject) {
+            try {
+                Object propVal = ((GroovyObject) expectedReceiver).getMetaClass().getProperty(expectedReceiver, expectedMethodName);
+                if (propVal instanceof Closure) {
+                    return;
+                }
+            } catch (MissingPropertyException mpe) {
+                // Do nothing - this just means the property didn't exist.
+            }
+        }
+        if (expectedReceiver instanceof Map && ((Map) expectedReceiver).containsKey(expectedMethodName)) {
+            Object mapVal = ((Map) expectedReceiver).get(expectedMethodName);
+            if (mapVal instanceof Closure) {
+                return;
```

Comment:
nit: I guess to be extra specific we could check that `"call".equals(methodName)` here and in the other condition before returning, or do we intentionally want to ignore closures entirely? I'm not really sure if `methodName` could be something other than `call` in this case, or if we gain anything by being more specific.

Reply:
If you're calling `foo.someField()`, where `someField` is a closure, you're calling `foo.someField.call()` - but it wouldn't hurt. I'll add something.